src/main/java/uk/gov/hmcts/reform/civil/service/SolicitorEmailValidation.java

```diff
@@ -0,0 +1,129 @@
+package uk.gov.hmcts.reform.civil.service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.model.OrganisationPolicy;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.common.Element;
+import uk.gov.hmcts.reform.civil.model.genapplication.GASolicitorDetailsGAspec;
+
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static uk.gov.hmcts.reform.civil.enums.YesOrNo.YES;
+import static uk.gov.hmcts.reform.civil.utils.ElementUtils.element;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class SolicitorEmailValidation {
+
+    private GASolicitorDetailsGAspec updateSolDetails(String updateEmail,
+                                                      GASolicitorDetailsGAspec generalAppSolicitor) {
+
+        GASolicitorDetailsGAspec.GASolicitorDetailsGAspecBuilder
+            gaSolicitorDetailsGAspecBuilder = GASolicitorDetailsGAspec.builder();
+
+        gaSolicitorDetailsGAspecBuilder.id(generalAppSolicitor.getId())
+            .forename(generalAppSolicitor.getForename())
+            .surname(generalAppSolicitor.getSurname())
+            .organisationIdentifier(generalAppSolicitor.getOrganisationIdentifier())
+            .email(updateEmail);
+
+        return gaSolicitorDetailsGAspecBuilder.build();
+
+    }
+
+    public GASolicitorDetailsGAspec checkIfOrgIDMatch(GASolicitorDetailsGAspec generalAppSolicitor,
+                                                       CaseData civilCaseData, CaseData gaCaseData) {
+
+        // civil claim applicant
+
```

Comment:
Instead of 3 blocks of repetitive code, the code can be refactored to have a method that minimizes this duplicates

Reply:
Done